packages/skill-lesson/trpc/trpc.server.ts

```diff
@@ -8,9 +8,29 @@
  * @see https://trpc.io/docs/v10/procedures
  */
 
-import {initTRPC} from '@trpc/server'
+import {TRPCError, initTRPC} from '@trpc/server'
 import superjson from 'superjson'
 import {type TrpcContext} from './trpc-context'
+import NextAuth, {DefaultSession} from 'next-auth'
+
+type Resolve<T> = T extends Function ? T : {[K in keyof T]: T[K]}
+
+type User = Resolve<
+  {
+    role?: string
+    purchases?: any[]
+  } & DefaultSession['user']
+>
+
+declare module 'next-auth' {
+  /**
+   * Returned by `useSession`, `getSession` and received as a prop on the
+   * `SessionProvider` React Context
+   */
+  interface Session {
+    user?: User
+  }
+}
```

Comment:
I don't totally have my head wrapped around the issue, but it seems like when this `declare module ...` was in the `next-auth.d.ts` at the root-level, it wasn't getting picked up by anything during a lint or build. In contrast to the `skill-api` package where it is getting picked up, I'm guessing it has something to do with it being transpiled as an [internal package](https://turbo.build/repo/docs/handbook/sharing-code/internal-packages).

Even adding `next-auth.d.ts` to the `include` of the `skill-lesson`'s `tsconfig.json` wasn't getting it picked up.

However, putting it directly in `trpc.server.ts` does get it picked up during build/lint.

What is extra strange is that the types were getting properly picked up by during development by VS Code.

![image](https://github.com/skillrecordings/products/assets/694063/a6bb236a-8194-4596-ad7e-84c32ed82630)